packages/grid/x-data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -179,1517 +56,1080 @@ describe('<DataGrid /> - Filter', () => {
       setProps({
-        initialState: {
-          filter: {
-            filterModel: {
-              items: [{ columnField: 'isPublished', operatorValue: 'is', value: 'false' }],
-            },
-          },
-        },
+        filterModel: { items: [{ columnField: 'voltage', operatorValue: 'is', value: 220 }] },
       });
-
-      expect(getColumnValues(0)).to.deep.equal(['Adidas', 'Puma']);
+      expect(getColumnValues(0)).to.deep.equal(['Hair Dryer', 'Microwave']);
     });
+  });
 
-    it('should allow to update the filters when initialized with initialState', () => {
-      render(
-        <Test
-          initialState={{
-            preferencePanel: {
-              open: true,
-              openedPanelValue: GridPreferencePanelsValue.filters,
-            },
-            filter: {
-              filterModel: {
-                items: [{ columnField: 'isPublished', operatorValue: 'is', value: 'true' }],
+  it('should translate operators dynamically in toolbar without crashing ', () => {
+    expect(() => {
+      return (
+        <div style={{ height: 400, width: '100%' }}>
+          <DataGrid
+            rows={[
+              {
+                id: 1,
+                quantity: 1,
               },
-            },
-          }}
-        />,
+            ]}
+            columns={[{ field: 'quantity', type: 'number', width: 150 }]}
+            filterModel={{
+              items: [
+                {
+                  columnField: 'quantity',
+                  id: 1619547587572,
+                  operatorValue: '=',
+                  value: '1',
+                },
+              ],
+            }}
+            components={{
+              Toolbar: GridToolbar,
+            }}
+          />
+        </div>
       );
+    }).not.to.throw();
+  });
 
-      expect(getColumnValues(0)).to.deep.equal(['Adidas', 'Puma']);
+  it('should apply the valueParser onto the filter value', () => {
+    render(
+      <TestCase
+        rows={[
+          {
+            id: 1,
+            amount: 0.5,
+          },
+          {
+            id: 2,
+            amount: 1,
+          },
+        ]}
+        columns={[
+          {
+            field: 'amount',
+            type: 'number',
+            valueParser: (value) => (value as number) / 100,
+          },
+        ]}
+        filterModel={{
+          items: [
+            {
+              columnField: 'amount',
+              operatorValue: '=',
+              value: 50,
+            },
+          ],
+        }}
+      />,
+    );
+    expect(getColumnValues(0)).to.deep.equal(['0.5']);
+  });
 
-      fireEvent.change(getByText(screen.getByRole('tooltip'), 'true').parentNode, {
-        target: { value: 'false' },
-      });
-      expect(getColumnValues(0)).to.deep.equal(['Nike']);
-    });
+  it('should include value-less operators when displaying the number of active filters', () => {
```

Comment:
Maybe should add the opposite test: "should not consider incomplete operators when displaying the number of active filters"